contracts/duckies/Gene.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+/*
+ * Gene is a number with a special structure that defines Duckie NFT traits.
+ * All traits are packed consequently in the reversed order in the Gene, meaning the first trait being stored in the last Gene bits.
+ * Each trait takes up the block of 8 bits in gene, thus having 256 possible values.
+ *
+ * Example of gene, following traits Class, Head and Body are defined:
+ *
+ * 00000001|00000010|00000011
+ *   Body    Head     Class
+ *
+ * This gene can be represented in uint24 as 66051.
+ * Traits have the following values: Body = 1, Head = 2, Class = 3.
+ */
+library Gene {
+    uint8 public constant BYTES_PER_TRAIT = 8;
+
+    enum Classes {
+        Common,
+        Rare,
+        Epic,
+        Legendary,
+        SuperLegendary,
+        Zombie
+    }
+
+    // order is important, see _generateGene()
+    enum Traits {
+        Class,
+        Body,
+        Head,
+        Background,
+        Element,
+        Eyes,
+        Beak,
+        Wings,
+        Firstname,
+        Lastname,
+        Temper,
+        Peculiarity
+    }
+
```

Comment:
how can we cover with tests?

Reply:
That generated values should be in the range [0, Max value].